pyoptsparse/pySNOPT/pySNOPT.py

```diff
@@ -676,30 +590,20 @@ def _set_snopt_options(self, iPrint, iSumm, cw, iw, rw):
         Set all the options into SNOPT that have been assigned
         by the user
         """
-
         # Set Options from the local options dictionary
         # ---------------------------------------------
         inform = np.array([-1], np.intc)
-        for name, value in self.set_options.items():
-            if name == "iprint" or name == "isumm":
+        for name, value in self.options.items():
+            # these do not get set using snset
+            if name in self.specialOptions or name in self.pythonOptions:
                 continue
 
             if isinstance(value, str):
-                if name == "start":
-                    if value == "Cold":
-                        snopt.snset("Cold start", iPrint, iSumm, inform, cw, iw, rw)
-                    elif value == "Warm":
-                        snopt.snset("Warm start", iPrint, iSumm, inform, cw, iw, rw)
-                elif name == "problem type":
-                    if value == "Minimize":
-                        snopt.snset("Minimize", iPrint, iSumm, inform, cw, iw, rw)
-                    elif value == "Maximize":
-                        snopt.snset("Maximize", iPrint, iSumm, inform, cw, iw, rw)
-                    elif value == "Feasible point":
-                        snopt.snset("Feasible point", iPrint, iSumm, inform, cw, iw, rw)
-                elif name == "print file":
+                if name == "Problem Type":
```

Comment:
I don't really get what is going on here, can you elaborate a bit? You are removing some calls to inner `snopt` functions so I want to double check

Reply:
I have removed the `Start` call here, since they were duplicate. The `Start` option can be set in one of two ways, either via the `Cold start` etc. option, or when calling `snkerc` (this one is mandatory and has always been done). I just removed this extra one here since it's not actually necessary.

Reply:
The `Problem Type` option has just been simplified, since we were setting the same string as before I just got rid of the if statements which were unnecessary. The code is functionally the same.

Reply:
I see the doubled `start` now, makes sense to just keep the `snkerc` one. I also agree with the `Problem Type` simplification, I am just missing where that option is defined now, I am probably messing between commits, but could you point me at that?

Reply:
Ahaha nevermind, I see you added it back in one of the last commits, all good here!